magenta-lib/src/main/scala/magenta/deployment_type/AutoScaling.scala

```diff
@@ -3,10 +3,14 @@ package magenta.deployment_type
 import magenta.{DeployTarget, DeploymentPackage, DeploymentResources, KeyRing}
 import magenta.tasks._
 
+sealed trait CdkTagRequirements
+case object MustBePresent extends CdkTagRequirements
+case object MustNotBePresent extends CdkTagRequirements
+
 object AutoScalingGroupLookup {
-  def getTargetAsgName(keyRing: KeyRing, target: DeployTarget, resources: DeploymentResources, pkg: DeploymentPackage) = {
+  def getTargetAsgName(keyRing: KeyRing, target: DeployTarget, cdkTagRequirements: Option[CdkTagRequirements], resources: DeploymentResources, pkg: DeploymentPackage) = {
```

Comment:
We could also use `cdkTagRequirements: Option[Boolean]` here, but I felt that the `sealed trait` helped to make things a little clearer.

Reply:
`Option[Boolean]` has three values too, which would be confusing 😄 .